### ISSUE TITLE
Feature request: algorithm improvement

### ISSUE THREAD

**Issue by dorseth** [zivester/node-quickthumb#20]
Hello, great lib, use it a lot, though maybe one thing can be improved:
let's say I have a 3000x3000px image, when the client with resolution of 1920px visits my site, the image is resized to 1920px, now let's say another client with resolution 1280x1024 visits the site,
what the lib does is - converts from the 3000x3000px image again, instead it could look into the cache folder and if there are any images that were converted to smaller resolutions then convert from them.

So basically the following should happen - 3000x3000 converted to 1920x1080, next client - 1920x1080 is converted to 1280x1024.
It would be great if it could be done, I think it might save a lot of server-side processing, especially on slow machines.

I would try to implement this myself, but at the moment I lack skill in pure Node. :)

**Comment by zivester** [comment]
Compressing and recompressing images is usually not a good idea.  Quality will suffer the more times you run any lossy algorithm (i.e. jpg) over and over again.  Using the original image is preferred.

Also, the library was meant as quick and dirty way to generate images on the fly.  For any production level code, I'd recommend preprocessing all images beforehand, and serving them up directly with a web server (nginx, apache, etc).

Thankful for the feedback and happy it is suiting your needs!
